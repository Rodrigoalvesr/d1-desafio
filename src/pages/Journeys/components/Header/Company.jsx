import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import { makeStyles } from '@material-ui/core/styles';
import Tooltip from '@Components/Tooltip';
import companyLogo from '@/assets/images/acme-logo.png';
import { withStyles } from '@material-ui/core/styles';

const DarkTooltip = withStyles({
  tooltip: {
    color: '#FFFFFF',
    backgroundColor: '#3E4157',
  },
  arrow: {
    color: '#3E4157',
  },
})(Tooltip);

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
  },
  avatar: {
    width: 24,
    height: 24,
    fontSize: 14,
    fontWeight: 'bold',
    background: theme.palette.primary.main,
  },
  avatarContent: {
    marginRight: 15,
  },
  companyContent: {
    display: 'flex',
    alignItems: 'center',
    height: 30,
    padding: '0 13px',
    border: '1px solid #EBEEF6',
    borderRadius: 5,
    background: '#FFFFFF',
  },
}));

function Company() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <div className={classes.avatarContent}>
        <DarkTooltip title="AntÃ´nio da Silva" placement="right" arrow>
          <Avatar className={classes.avatar}>A</Avatar>
        </DarkTooltip>
      </div>
      <div className={classes.companyContent}>
        <img alt="acme" src={companyLogo} height={26} />
      </div>
    </div>
  );
}

export default Company;
