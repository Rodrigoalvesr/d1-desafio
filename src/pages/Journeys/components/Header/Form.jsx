import React from 'react';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import SearchIcon from '@material-ui/icons/Search';
import { useDispatch } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import Input from '@Components/Input';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    alignItems: 'center',
    '& > *': {
      marginRight: 15,
    },
  },
  input: {
    width: 230,
  },
});

function Form() {
  const classes = useStyles();
  const dispatch = useDispatch();

  const handleOpenModal = React.useCallback(() => {
    dispatch({ type: 'JOURNEY_OPEN_MODAL' });
  }, [dispatch]);

  return (
    <div className={classes.root}>
      <Input
        placeholder="Buscar"
        startIcon={<SearchIcon fontSize="small" />}
        className={classes.input}
      />
      <Button
        color="primary"
        variant="contained"
        disableElevation
        startIcon={<AddIcon />}
        onClick={handleOpenModal}
      >
        Nova Jornada
      </Button>
    </div>
  );
}

export default Form;
