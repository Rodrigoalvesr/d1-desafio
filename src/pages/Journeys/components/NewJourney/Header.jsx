import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    paddingBottom: 15,
    marginBottom: 15,
    borderBottom: '1px solid #E4E6F1',
  },
  title: {
    color: '#9196AB',
    fontSize: 14,
    fontWeight: 'bold',
  },
});

function Header() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Typography className={classes.title}>Nova Jornada</Typography>
    </div>
  );
}

export default Header;
