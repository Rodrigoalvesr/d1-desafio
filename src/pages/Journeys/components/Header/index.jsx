import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Company from './Company';
import { useDispatch } from 'react-redux';
import Form from './Form';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: 90,
    padding: '0 90px',
  },
});

function Header() {
  const classes = useStyles();
  const dispatch = useDispatch();

  const handleOpenModal = React.useCallback(() => {
    dispatch({ type: 'JOURNEY_OPEN_MODAL' });
  }, [dispatch]);

  return (
    <header className={classes.root}>
      <Company />
      <Form handleOpenModal={handleOpenModal} />
    </header>
  );
}

export default Header;
