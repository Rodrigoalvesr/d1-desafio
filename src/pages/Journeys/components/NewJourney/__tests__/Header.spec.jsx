import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

describe('Header', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Header />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
