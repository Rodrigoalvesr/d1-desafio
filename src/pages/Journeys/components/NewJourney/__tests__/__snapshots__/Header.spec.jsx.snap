// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root makeStyles-title-2 MuiTypography-body1"
  >
    Nova Jornada
  </p>
</div>
`;
